pkg/gameserverallocations/allocator_test.go

```diff
@@ -81,6 +81,7 @@ func TestAllocatorAllocate(t *testing.T) {
 
 	require.NoError(t, a.Run(ctx))
 	// wait for it to be up and running
+	// nolint:staticcheck
 	err := wait.PollImmediate(time.Second, 10*time.Second, func() (done bool, err error) {
```

Comment:
I saw you fixed this lint warning with `PollUntilContextTimeout` which looks correct. Do you mind fix these too? We can also do this in a separate PR. Up to you.

Reply:
sure, will replace all `PollImmediate` with `PollUntilContextTimeout`.

Reply:
I've replaced `PollImmediate` with `PollUntilContextTimeout`.
Is it fine to replace `wait.Poll(time.Second, 10*time.Second, func() (bool, error)` with `wait.PollUntilContextTimeout(ctx, time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error))`?

Reply:
Yes, `wait.Poll` can be replaced with `wait.PollUntilContextTimeout`. Again, up to you whether do that in this PR or a following up PR.